server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/controllers/DatasetController.java

```diff
@@ -323,6 +323,11 @@ public Dataset ingestDataset(@ApiParam(value = "JSON representing an input datas
             throw new BadRequestException("Invalid input dataset. Please verify that the dataset does not contain any wrong parameters.");
         }
 
+        if (JsonUtils.extractValueFromJsonString("id", inDatasetJson) != ""){
+            throw new IncoreHTTPException(Response.Status.BAD_REQUEST, "Ids are auto-generated by the system. " +
```

Comment:
is there any reason why you are not using `UserInfoUtils.throwExceptionIfIdPresent`?